projects/observability/src/shared/components/cartesian/chart-sync-service.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@angular/core';
+import { Observable, Subject } from 'rxjs';
+import { filter, switchMap } from 'rxjs/operators';
+import { Band, CartesianSelectedData, Series } from '../../../public-api';
+import { MouseLocationData } from '../utils/mouse-tracking/mouse-tracking';
+
+@Injectable({ providedIn: 'root' })
+export class ChartSyncService<TData> {
+  private readonly locationChangeSubject: Subject<
+    MouseLocationData<TData, Series<TData> | Band<TData>>[] | CartesianSelectedData<TData>
+  > = new Subject();
+
+  private groupId?: string;
```

Comment:
No need to store state in two places even if correct, it's making assumptions about ordering that aren't necessary. It's cleaner to hold this inside the subject e.g. `{groupId: string, locationData: ... }`, then filter against the pipeline value rather than reaching into service state as line 28 is currently doing.